libcore/src/events/TrainEvent.cpp

```diff
@@ -0,0 +1,320 @@
+#include "TrainEvent.h"
+
+#include "geometry/SubRoom.h"
+#include "geometry/Wall.h"
+
+void TrainEvent::Process()
```

Comment:
I think the event handler should be implemented separately. I would like to have a very small event object and an additional event handler object. When an event is triggered, the corresponding event handler should be called for this event. This also avoids that the event objects are bloated. Maybe we should do this in another PR/Issue. This would be too much and not focused on this PR.

Reply:
Created issue: #764